app/src/main/kotlin/com/intive/picover/parties/viewmodel/PartiesViewModel.kt

```diff
@@ -19,15 +23,28 @@ import kotlinx.coroutines.launch
 @HiltViewModel
 class PartiesViewModel @Inject constructor(
 	private val partiesRepository: PartiesRepository,
-) : ViewModel() {
+) : MVIViewModel<PartiesEvent>(),
+	SideEffectEmitter<PartiesSideEffect> by SideEffectEmitterImplementation() {
```

Comment:
Orbit mvi library is doing the following:
- ViewModel implements ContainerHost<State, SideEffect> interface
- ContainerHost have Container val inside
- in viewmodel, we HAVE to override container val, using ViewModel container extension function
- …whichUse viewModelScope.container fun to create the RealContainer object, in which we have all the values like state, effects etc. 
- there are extension functions for ContainerHost like intent, reduce, postSetEffect which handles the container
Looks quite complicated.. :D